chunky/src/java/se/llbit/math/Octree.java

```diff
@@ -425,26 +425,79 @@ public boolean isInside(Vector3 o) {
    * @return {@code false} if the ray doesn't intersect the octree.
    */
   private boolean enterOctree(Ray ray) {
-    double nx, ny, nz;
+    double nx = 0, ny = 0, nz = 0;
     double octree_size = 1 << getDepth();
 
+    double tMin = Double.NEGATIVE_INFINITY;
+    double tMax = Double.POSITIVE_INFINITY;
+
+    // Note: The following is made to be robust against edge cases (infinity/NaN)
+    // without additional branches, be careful when editing.
+
+    // Explanation:
+    // the ray can have its x coordinate be 0, which mean that invDirX is Infinity.
+    // This is not a problem as math works out well in this case:
+
+    // If we have 0 < ray.o.x < octree_size, meaning the ray is correctly aligned in
+    // x to intersect with the octree, [tXMin ; tXMax] will have the values
+    // [-Infinity ; Infinity] meaning its union with the interval [tMin ; tMax]
+    // will keep the value [tMin ; tMax].
+
+    // On the other hand if ray.o.x < 0 or if ray.o.x > octree_size then both
+    // tXMin and tXMax will have the value -Infinity (resp. +Infinity)
+    // Meaning the interval [tXMin ; tXMax] (and [tMin ; tMax] once it is updated to be the union)
+    // will be reduced to a single values: -Infinity (resp. +Infinity)
+    // (this description is not mathematically rigorous as infinity is not really a value
+    // but is enough here).
+    // As ray.d is not a null vector, at least one of its component is not 0
+    // and will have an associated interval with finite bounds, meaning the union of those interval
+    // will give an empty set (in practice this is the condition
```

Comment:
A union with an interval never gives an empty set. Do you mean _intersection_ instead of _union_ here? Same for the first paragraph? :thinking:

Reply:
You're right I mixed up, it's intersection everywhere